rfc/0003-policy-signing.md

```diff
@@ -0,0 +1,440 @@
+|              |                                  |
+| :----------- | :------------------------------- |
+| Feature Name | Policy Signing                   |
+| Start Date   | 21 Jan 2022                      |
+| Category     | Security                         |
+| RFC PR       | [#147](https://github.com/kubewarden/kubewarden-controller/pull/147)  |
+
+# Summary
+[summary]: #summary
+
+> Brief (one-paragraph) explanation of the feature.
+
+We want to provide a way to verify the integrity and authenticity of our policies, so that end-users can trust them.
+
+To accomplish that, we will provide ways to sign and verify policies.
+
+
+# Motivation
+[motivation]: #motivation
+
+> - Why are we doing this?
+> - What use cases does it support?
+> - What is the expected outcome?
+>
+> Describe the problem you are trying to solve, and its constraints, without coupling them too closely to the solution you have in mind. If this RFC is not accepted, the motivation can be used to develop alternative solutions.
+
+Our policies are binary blobs, that makes hard to ensure the policy we're about to run
+matches with the source code tagged inside of a git repository.
+
+This is what we want to achieve with this proposal:
+
+* policy authors: have a way to sign their policies, to attest their provenance.
+* end users: have a way to verify the integrity of a policy that they want to load inside of their cluster. Allow them to execute only policies they trust.
+
+## Examples / User Stories
+[examples]: #examples
+
+> Examples of how the feature will be used. Interactions should show the action and the response. When appropriate, provide user stories in the form of "As a [role], I want [feature], so [that]."
+
+### Execute only policies coming from trusted parties
+
+> As a Kubernates administrator,
+> I want to enforce only Kubewarden policies that have been produced by authors I trust to be loaded.
+
+### Execute only policies that have been approved by selected people
+
+> As a group of Kubernetes administrators,
+> we want to deploy in production only policies that we have approved.
+
+In that case, the Kubernetes administrators would:
+
+  * Download a pre-built Kubewarden policy
+  * Verify it's being signed by its author
```

Comment:
Verification shouldn't be an extra step for the admin. `zypper` and the open build service (OBS) is a good role model here. The OBS does all the key generation and signing of repos, no extra tasks for software authors. Adding a repo is where trust is verified. Package installation from a trusted repo never asks about trust.

Policy signing should follow a similar model. Explicit verification when a policy registry gets added, implicit (hidden) verification when a policy is downloaded.